webrtc/browser-to-browser.md

```diff
@@ -0,0 +1,119 @@
+# WebRTC browser-to-browser
+
+| Lifecycle Stage | Maturity      | Status | Latest Revision |
+|-----------------|---------------|--------|-----------------|
+| 1A              | Working Draft | Active | r0, 2022-12-15  |
+
+## Motivation
+
+**Hole punching in the browser**
+
+Browser _A_ wants to connect to Browser node _B_ with the help of server node _R_.
+Both _A_ and _B_ can not listen for incoming connections due to the restriction of the browser platform and being behind a NAT and/or firewall.
+Note that _A_ and/or _B_ may as well be non-browser nodes behind NATs and/or firewalls.
+However, for two non-browser nodes using TCP or QUIC hole punching with [DCUtR] will be the more efficient way to establish a direct connection.
+
+On a historical note, this specification replaces the existing [libp2p WebRTC star](https://github.com/libp2p/js-libp2p-webrtc-star) and [libp2p WebRTC direct](https://github.com/libp2p/js-libp2p-webrtc-direct) protocols.
+
+## Connection Establishment
+
+1. _B_ advertises support for the WebRTC browser-to-browser protocol by appending `/webrtc-direct` to its relayed multiaddr e.g. `/ip6/fe80::883:a581:fff1:833/udp/4001/quic/p2p/<relay-peer-id>/p2p-circuit/webrtc-direct/p2p/<b-peer-id>`.
```

Comment:
My understanding of our chat today was that this would be a separate multiaddr. B would advertise:

`/ip6/.../p2p-circuit/b-peerID`, `/webrtc-direct/b-peerID`, and any other relay addrs. Not that it would sent double the multiaddrs (i.e. `/ip6/.../p2p-circuit/b-peerID` and `/ip6/.../p2p-circuit/webrtc-direct/b-peerID`,)

Reply:
Aside, not relevant but a quic multiaddr doesn't really make sense in the browser to browser example since a browser can't connect to the relay's quic address. In reality this would probably look something like 

B's multiaddrs:
1. `/ip6/.../quic/p2p/relay-peer-id/p2p-circuit/b-peer-id`
2. `/ip6/.../wss/p2p/relay-peer-id/p2p-circuit/b-peer-id`
3. `/webrtc-direct/b-peer-id`

Reply:
Yes. See rational in https://github.com/libp2p/specs/pull/497#discussion_r1113332002. Let's continue the discussion there if you don't mind @MarcoPolo. Hope that thread has not yet collapsed on your view.

Reply:
> Aside, not relevant but a quic multiaddr doesn't really make sense in the browser to browser example since a browser can't connect to the relay's quic address. In reality this would probably look something like

Good point. 165bd52 uses webtransport instead of plain quic for the relay.

Reply:
> Yes. See rational in [#497 (comment)](https://github.com/libp2p/specs/pull/497#discussion_r1113332002). Let's continue the discussion there if you don't mind @MarcoPolo. Hope that thread has not yet collapsed on your view.

I just closed the above linked discussion. In case you feel strongly about multiple multiaddresses @MarcoPolo please open a new thread. Resolving here as well.